### PR TITLE
doc: 3025 http keyword updates v4

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3025

Describe changes:
- applied feedback from https://github.com/OISF/suricata/pull/10329
- updated all http keywords documentation
- RTD build of branch https://rtd-test-suricata.readthedocs.io/en/latest/rules/http-keywords.html

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -1,813 +1,885 @@
...
-http.start                     http_start (*)           Both
-http.protocol                  http_protocol (*)        Both
-http.header_names              http_header_names (*)    Both
-============================== ======================== ==================
-
-\*) sticky buffer
+Using the HTTP specific sticky buffers (see :ref:`rules-modifiers`) provides a
+way to efficiently inspect the specific fields of HTTP protocol communications.
+After specifying a sticky buffer in a rule it should be followed by one or
+more :doc:`payload-keywords`.
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:13]: I've noticed that the notes about corresponding PCRE modifiers were removed. Was wondering if adding just one mention to that, maybe in this section, pointing to Suricata's modifiers (https://docs.suricata.io/en/latest/rules/payload-keywords.html#suricata-s-modifiers)  would be a good idea?

File: doc/userguide/rules/http-keywords.rst [hunk 2]
```diff
@@ -823,38 +895,365 @@ Flash decompression must be enabled under ``libhtp`` configuration:
...
+  content:"User-Agent|3a 20|Mozilla/5.0|0d 0a|"; \
+  content:"User-Agent|3a 20|Chrome|0d 0a|";` classtype:bad-unknown; sid:73; rev:1;)
+
+.. _http.header_names:
+
+http.header_names
+-----------------
+
+The ``http.header_names`` keyword is used to match on the names of the headers
+in an HTTP request or response. This is useful for checking for a headers
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:1127]: nit: 
```suggestion
in an HTTP request or response. This is useful for checking for a header's
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
Hi there, just reviewed this comparing to comments from the previous PR, and resolved all comments by me that seemed to have been addressed. Seems that we're mostly done! 🎉  💪🏽 
I think the other requests were also tackled.
I've left inline comments for missed typos and for a possible note to be added about PCRE modifiers.

Commit separation was rather strict! Maybe they could have been squashed as per change type (removal / reordering etc), but from my part I don't think it's necessary to change that, now. Dunno if others would have a comment on this.

I've noticed that the keywords reordering was done based on alphabetical + if the keyword is applied to request/ response or both. Seems to work for me, too.

Please let us know if there is anything in particular that was not checked and you want to ensure that was reviewed!

(thanks for this work, as usual! :) )

**Comment by jmtaylor90** [comment]
continued in https://github.com/OISF/suricata/pull/10735

**Pull request closed without merging**